### ISSUE TITLE
JAVA_TOOL_OPTIONS overwrite when using gradle build

### ISSUE THREAD

**Issue by mkushakov** [jenkinsci/datadog-plugin#442]
**Describe the bug**
When running DataDog-plugin for jenkins `7.1.1` using gradle job is failing with following error:
```
+ ./gradlew -i -Ddebuginfo=true -PbranchName=main --refresh-dependencies clean infoGit infoExtension build -x test
 Picked up JAVA_TOOL_OPTIONS: -javaagent:/var/lib/jenkins/workspace/repo_name_main/.datadog/dd-java-agent.jar
 Error opening zip file or JAR manifest missing : /var/lib/jenkins/workspace/repo_name_main/.datadog/dd-java-agent.jar
 Error occurred during initialization of VM
 agent library failed to init: instrument
```

Looks like related to #433 changes. I don't think it is good idea to overwrite `JAVA_TOOL_OPTIONS` as it is common for many process, but instead it should be appended.

**Expected behavior**
Normal build log:
```
+ ./gradlew -i -Ddebuginfo=true -PbranchName=main --refresh-dependencies clean infoGit infoExtension build -x test
 Initialized native services in: /var/lib/jenkins/.gradle/native
 Initialized jansi services in: /var/lib/jenkins/.gradle/native
 Received JVM installation metadata from '/usr/lib/jvm/java-17-amazon-corretto': {JAVA_HOME=/usr/lib/jvm/java-17-amazon-corretto, JAVA_VERSION=17.0.10, JAVA_VENDOR=Amazon.com Inc., RUNTIME_NAME=OpenJDK Runtime Environment, RUNTIME_VERSION=17.0.10+8-LTS, VM_NAME=OpenJDK 64-Bit Server VM, VM_VERSION=17.0.10+8-LTS, VM_VENDOR=Amazon.com Inc., OS_ARCH=amd64}
``` 

**Additional context**
Works perfectly on `7.1.0`

**Comment by nikita-tkachenko-datadog** [comment]
Hi @mkushakov, thank you for reporting this!
Could you please provide some additional details about the pipeline that runs the Gradle job?
Is it correct to assume that you have [enabled Test Visibility](https://github.com/jenkinsci/datadog-plugin?tab=readme-ov-file#test-visibility-configuration) in the pipeline's settings?
If you are setting `JAVA_TOOL_OPTIONS` anywhere, could you please let me know how exactly you're doing it?
Thanks.

**Comment by mkushakov** [comment]
Hello @nikita-tkachenko-datadog
Thanks for fast reply and sorry for reduced context.
Yes, we are using DD Test Visibility:
```
    options {
        datadog(tags: ["team:name"], testVisibility: [ enabled: true,  serviceName: "my-service", languages: ["JAVA"] ])
```
as for `JAVA_TOOL_OPTIONS` i need to dig to the build, as it is not defined explicitly, but may be set by some other tool or plugin

**Comment by mkushakov** [comment]
I have done some debugging, and it seems my first conclusion was wrong. The issue is not related to `JAVA_TOOL_OPTIONS` but rather availbility of `.datadog/dd-java-agent.jar`
When debugging I was able to see that on good build i have following logs:
```
[datadog] Configuring DD Java tracer: tracer installed in /var/lib/jenkins/workspace/repo_name_main on org.jenkinsci.plugins.vSphereCloudProvisionedSlave[worker-name-1]
[Pipeline] stage
[Pipeline] { (Declarative: Checkout SCM)
[datadog] Configuring DD Java tracer: using existing tracer available at /var/lib/jenkins/workspace/repo_name_main/.datadog/dd-java-agent.jar
[Pipeline] checkout
```
while on failed build i have only:
```
 [datadog] Configuring DD Java tracer: tracer installed in /var/lib/jenkins/workspace/repo_name_main on org.jenkinsci.plugins.vSphereCloudProvisionedSlave[worker-name-1]
 [Pipeline] stage
 [Pipeline] { (Declarative: Checkout SCM)
 [Pipeline] checkout
```

We are using dynamic workers nodes, so we are spawning new VM on request and reuse it till no jobs are scheduled for 60 minutes, then we destroy VM.
When fresh job is spawned on VM - it was able to dynamically fetch and install `dd-java-agent.jar` then we have `cleanWS()` at the end of the build to clean workspace directory. So next time job will run on the same worker it will fail with this error:
```
 Error opening zip file or JAR manifest missing : /var/lib/jenkins/workspace/repo_name_main/.datadog/dd-java-agent.jar
 Error occurred during initialization of VM
 agent library failed to init: instrument
```
I was able to reproduce this issue on `7.1.0` but on `7.0.1` it always works. Not sure what changes has been done on `7.1.0` regarding validating if agent is present or not on the host.

**Comment by nikita-tkachenko-datadog** [comment]
Many thanks for these details, they help a lot! I think I understand what the issue is now.

Before installing the tracer the instrumentation logic checks if installation has already been performed for a given host/pipeline. Rather than checking the actual files on disk, it looks for in-memory metadata that is "attached" to the pipeline object during the tracer installation. That metadata is checked to see if there's record of installing the tracer on a specific host.

What changed in 7.1.0 was the way the plugin determines hostname for Jenkins nodes. I assume that prior to that release the plugin could not reliably detect the names of your dynamic VMs, which is why the installation was performed every time. Now that the names are detected correctly the installation is skipped but, since you're doing some cleanup, the actual tracer library is not there.

I will update the logic to verify that the library is present before skipping the installation.

**Comment by nikita-tkachenko-datadog** [comment]
@mkushakov the fix was released in the plugin 7.1.3. Please let me know if the problem is still there in the new version. Thanks!
